apps/prairielearn/src/tests/permissions/institutionAdminAccess.test.ts

```diff
@@ -0,0 +1,134 @@
+import { assert } from 'chai';
+import { step } from 'mocha-steps';
+import fetch from 'node-fetch';
+
+import { queryAsync, queryRow } from '@prairielearn/postgres';
+
+import { config } from '../../lib/config.js';
+import { UserSchema } from '../../lib/db-types.js';
+import { selectUserByUid } from '../../models/user.js';
+import * as helperServer from '../helperServer.js';
+import { withUser } from '../utils/auth.js';
+
+const SITE_URL = `http://localhost:${config.serverPort}`;
+const INSTITUTION_ADMIN_COURSES = `${SITE_URL}/pl/institution/1/admin/courses`;
+const COURSE_URL = `${SITE_URL}/pl/course/1/course_admin/instances`;
+const COURSE_INSTANCE_URL = `${SITE_URL}/pl/course_instance/1/instructor/instance_admin/assessments`;
+const ASSESSMENT_INSTANCES_URL = `${SITE_URL}/pl/course_instance/1/instructor/assessment/1/instances`;
+
+interface AuthUser {
+  name: string;
+  uid: string;
+  uin?: string;
+  isAdministrator?: boolean;
+}
+
+const ADMIN_USER = {
+  name: 'Admin',
+  uid: 'admin@example.com',
+  uin: 'admin',
+  isAdministrator: true,
+};
+
+const INSTITUTION_ADMIN_USER = {
+  name: 'Institution Admin',
+  uid: 'institution-admin@example.com',
+  uin: 'institution-admin',
+};
+
+async function insertUser(user: AuthUser) {
+  const newUser = await queryRow(
+    `INSERT INTO users (name, uid) VALUES ($name, $uid) RETURNING *;`,
+    {
+      name: user.name,
+      uid: user.uid,
+    },
+    UserSchema,
+  );
+
+  if (user.isAdministrator) {
+    await queryAsync(`INSERT INTO administrators (user_id) VALUES ($user_id);`, {
+      user_id: newUser.user_id,
+    });
+  }
```

Comment:
At some point (doesn't have to be in this PR) it would be good to add `insertUser()` to `models/user` and similarly for `administrators`.

Reply:
Ack; I'll save that for a future PR.